src/OrleansTelemetryConsumers.Counters/PerfCounterTelemetryConsumer.cs

```diff
@@ -0,0 +1,224 @@
+            foreach (PerfCounterConfigData cd in perfCounterData)
+            {
+                var perfCounterName = GetPerfCounterName(cd);
+                var description = cd.Name.Name;
+
+                var msg = string.Format("Registering perf counter {0}", perfCounterName);
+                Console.WriteLine(msg);
+
+                ctrCreationData.Add(new CounterCreationData(perfCounterName, description, PerformanceCounterType.NumberOfItems32));
+            }
+            return ctrCreationData.ToArray();
+        }
+
+        /// <summary>
+        /// Register Orleans perf counters with Windows
+        /// </summary>
+        /// <remarks>Note: Program needs to be running as Administrator to be able to delete Windows perf counters.</remarks>
+        public void InstallCounters()
+        {
+            var collection = new CounterCreationDataCollection();
+            collection.AddRange(GetCounterCreationData());
+
+            PerformanceCounterCategory.Create(
+                CATEGORY_NAME,
+                CATEGORY_DESCRIPTION,
+                PerformanceCounterCategoryType.SingleInstance,
+                collection);
+        }
+
+        /// <summary>
+        /// Delete any existing perf counters registered with Windows
+        /// </summary>
+        /// <remarks>Note: Program needs to be running as Administrator to be able to delete Windows perf counters.</remarks>
+        public void DeleteCounters()
+        {
+            PerformanceCounterCategory.Delete(CATEGORY_NAME);
+        }
+
+        private PerfCounterConfigData GetCounter(string counterName)
+        {
+            return perfCounterData.Where(pcd => GetPerfCounterName(pcd) == counterName).SingleOrDefault();
+        }
+
+        #endregion
+
+        #region IMetricTelemetryConsumer Methods
+
+        public void IncrementMetric(string name) => WriteMetric(name, UpdateMode.Increment);
+
+        public void IncrementMetric(string name, double value) => WriteMetric(name, UpdateMode.Increment, value);
+
+        public void TrackMetric(string name, double value, IDictionary<string, string> properties = null) => WriteMetric(name, UpdateMode.Set, value);
+
+        public void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null) => WriteMetric(name, UpdateMode.Set, value.Ticks);
+
+        public void DecrementMetric(string name) => WriteMetric(name, UpdateMode.Decrement);
+
+        public void DecrementMetric(string name, double value) => WriteMetric(name, UpdateMode.Decrement, value);
+
+        public void Flush() { }
+
+        public void Close() { }
+
+        private void WriteMetric(string name, UpdateMode mode = UpdateMode.Increment, double? value = null)
+        {
+            PerfCounterConfigData cd = GetCounter(name);
+            if (cd == null)
+            {
+                if (logger.IsVerbose) logger.Verbose(ErrorCode.PerfCounterNotFound, "No perf counter found for {0}", name);
```

Comment:
for any metric not listed in Orleans runtime performance counters, this will log incessantly in verbose mode.  This is far more common than in the original case.